usd/src/sdf_parser/Joint.cc

```diff
@@ -0,0 +1,413 @@
+        "specifying joint axis unit vectors.")));
+      return errors;
+    }
+
+    // Revolute joint limits in SDF are in radians, but USD expects degrees
+    // of C++ type float
+    auto sdfLimitDegrees = static_cast<float>(IGN_RTOD(axis->Lower()));
+    usdJoint.CreateLowerLimitAttr().Set(sdfLimitDegrees);
+    sdfLimitDegrees = static_cast<float>(IGN_RTOD(axis->Upper()));
+    usdJoint.CreateUpperLimitAttr().Set(sdfLimitDegrees);
+
+    pxr::UsdPrim usdJointPrim = _stage->GetPrimAtPath(pxr::SdfPath(_path));
+
+    auto drive =
+      pxr::UsdPhysicsDriveAPI::Apply(usdJointPrim, pxr::TfToken("angular"));
+    if (!drive)
+    {
+      errors.push_back(UsdError(sdf::usd::UsdErrorCode::FAILED_PRIM_API_APPLY,
+            "Internal error: unable to mark link at path [" + _path +
+            "] as a UsdPhysicsDriveAPI."));
+      return errors;
+    }
+
+    // TODO(ahcorde) Review damping and stiffness values
+    double damping = axis->Damping();
+    drive.CreateDampingAttr().Set(static_cast<float>(damping));
+    double stiffness = axis->Stiffness();
+    drive.CreateStiffnessAttr().Set(static_cast<float>(stiffness));
+    drive.CreateMaxForceAttr().Set(static_cast<float>(axis->Effort()));
+
+    return errors;
+  }
+
+  /// \brief Helper function to parse a SDF prismatic joint to its USD
+  /// representation.
+  /// \param[in] _joint The SDF joint to parse
+  /// \param[in] _stage The stage that _joint should belong to. This must be a
+  /// valid, initialized stage.
+  /// \param[in] _path The path in _stage where the USD representation of _joint
+  /// will be defined.
+  /// \return UsdErrors, which is a vector of UsdError objects. Each UsdError
+  /// includes an error code and message. An empty vector indicates no error
+  /// occurred when converting _joint to its USD representation.
+  UsdErrors ParseSdfPrismaticJoint(const sdf::Joint &_joint,
+      pxr::UsdStageRefPtr &_stage, const std::string &_path)
+  {
+    UsdErrors errors;
+
+    auto usdJoint =
+      pxr::UsdPhysicsPrismaticJoint::Define(_stage, pxr::SdfPath(_path));
+
+    const auto axis = _joint.Axis();
+
+    if (axis->Xyz() == ignition::math::Vector3d::UnitX ||
+        axis->Xyz() == -ignition::math::Vector3d::UnitX)
+    {
+      usdJoint.CreateAxisAttr().Set(pxr::TfToken("X"));
+    }
+    else if (axis->Xyz() == ignition::math::Vector3d::UnitY ||
+        axis->Xyz() == -ignition::math::Vector3d::UnitY)
+    {
+      usdJoint.CreateAxisAttr().Set(pxr::TfToken("Y"));
+    }
+    else if (axis->Xyz() == ignition::math::Vector3d::UnitZ ||
+        axis->Xyz() == -ignition::math::Vector3d::UnitZ)
+    {
+      usdJoint.CreateAxisAttr().Set(pxr::TfToken("Z"));
+    }
+    else
+    {
+      errors.push_back(UsdError(sdf::Error(sdf::ErrorCode::ELEMENT_INVALID,
+        "Prismatic joint [" + _joint.Name() + "] has specified an axis "
+        "[x y z], but USD only supports integer values of [0, 1] when"
```

Comment:
repeat: I meant to suggest printing `axis->Xyz()` in place of `x y z` in this message

Reply:
https://github.com/ignitionrobotics/sdformat/pull/837/commits/c03c363adaa6e469a66baab012c1ad6a0c981dbc